fem/fespace.hpp

```diff
@@ -112,6 +112,8 @@ class FiniteElementSpace
 
    mutable Table *elem_dof; // if NURBS FE space, not owned; otherwise, owned.
    Table *bdrElem_dof; // used only with NURBS FE spaces; not owned.
+   Table *face_dof; // used only with NURBS FE spaces;
+   Array<int> face_to_be; // used only with NURBS FE spaces;
```

Comment:
Do these member data belong in `FiniteElementSpace` or in `NURBSExtension`?

Because they are going to be NULL unless the mesh is a NURBS mesh, placing them in `NURBSExtension` might be more logical?

It may also simplify some of the constructor/destructor logic, which seems to me to be a bit convoluted.

Reply:
I have considered them placing them there just as `elem_dof` and `elemBdr_dof`, this would have been required if the `face_dofs` where generated in the NURBSExt (just as the other Tables). However, this is not the case, the current mechanism requires fespace and mesh data in order to generate the `face_dof` table and `face_to_be` mapping.

Perhaps the comment should say : // currently only used with NURBS FE spaces
Both data members could be used by  fespace types other the NURBS in the future.

So I think it is both natural and convienent to have them in the fespace.

Reply:
I agree it is constructor/destructor convoluted, but it only becomes worse when it is changed IMO.

The NURBS patch topology is technically a mesh, but all knotspans also provide a mesh (more aligned with the standard notion of a mesh). When calling refine, other constructor/update paths are taken. The fespace for the solution can create its own NURBSext or use the one from the `Mesh->Nodes()` fespace. This makes it non-trivial.
Perhaps a mayor overhaul of NURBS initialization is required at some point.